### ISSUE TITLE
Get some values of an offline client

### ISSUE THREAD

**Issue by realKitsune** [planetteamspeak/ts3phpframework#85]
Hello,

I've been trying to display "lastconnected" on my webconsole.
I've looked the [Teamspeak Server Query Manual Page 26](http://media.teamspeak.com/ts3_literature/TeamSpeak%203%20Server%20Query%20Manual.pdf#page=26) and found what to use but it seems to not working (it doesn't get any value). I'm using clientListDb().
Here's a part of the code concerned:

```
foreach($groupSprAdm_List as $ts3_GroupList)
				{
					$getUID = array_search($ts3_GroupList['client_unique_identifier'], $groupSprAdm_UniqueId);
					$getName = $groupSprAdm_WebNames[$getUID];
					$getDbInfo = $ts3_VirtualServer->clientListDb($ts3_GroupList['cldbid']);

					if($ts3_VirtualServer->clientList(array("client_database_id" => $ts3_GroupList['cldbid']))){
	 					$connected = "<span style='color:rgb(198,239,206)'>• Connecté</span>"; 
					}else{
	 					$connected = "<span style='color:rgb(255,199,206)'>• Non connecté, ".date("d/m/Y", $getDbInfo['client_lastconnected'])."</span>"; 
					}

  				
					echo "<tr>";
					echo "<td rowspan=2 style=\"color:rgb(255,199,206);text-shadow: 2px 2px 3px rgba(156,0,6,0.8); width:40%;\"> <img src=\"/images/groups/18.png\" style=\"vertical-align:text-bottom\"> <b>".$ts3_GroupList['client_nickname']."</b> (#".$ts3_GroupList['cldbid'].")</td>";
					echo "<td><b style=\"color:rgb(255,199,206)\";>UniqueID: </b><code>".$ts3_GroupList['client_unique_identifier']."</code> <b style=\"color:rgb(255,199,206)\";>User: </b> <code> ".$getName."</code></td>";
					echo "</tr>";
					echo "<tr>";
					echo "<td>".$connected."</td>";
					echo "</tr>";
				}
```

What I get:
• Non connecté, 01/01/1970
What I should get (since I disconnected today from my Teamspeak):
• Non connecté, 20/06/2018 

Thanks in advance for help

**Comment by ronindesign** [comment]
01/01/1970 is unix epoch, so probably somewhere in your code you're setting/getting null or 0 value for the date. Let me review and see if I can tell where this is happening.

**Comment by ronindesign** [comment]
Looks like there is a bug as of TeamSpeak server 3.2.0. Several people reporting your same issue.
Ref: https://forum.teamspeak.com/threads/135919-Diferentlastconnected

We'll keep an eye on this and see how it shakes out.

**Comment by realKitsune** [comment]
Hi,

Using this:
$getDbInfo = $ts3_VirtualServer->**clientInfoDb**($ts3_GroupList['cldbid']);

Instead of:
$getDbInfo = $ts3_VirtualServer->**clientListDb**($ts3_GroupList['cldbid']);

Solved my problem. I can get offline clients values such as client_description, client_lastconnected (which seems to be correct values), etc.

**Comment by realKitsune** [comment]
Hi,

It appeared that some values were not correct. I realized it right now, for some reason it gave the accurate date of last login for some users, but mine is not correct (says 21/06/18 instead of 25-26/06/18 like others).
So seems that I have the same problem.

```
root@ST8-Kitsune:~# php --version
PHP 5.4.45-0+deb7u13 (cli) (built: Mar 29 2018 16:33:34)
Copyright (c) 1997-2014 The PHP Group
Zend Engine v2.4.0, Copyright (c) 1998-2014 Zend Technologies
```

**Comment by ronindesign** [comment]
Yeah, if I remember the only things that might effect lastconnected:

- The bug in forum thread mentioned above.
- Desync (database info doesn't match current client info)
- The DB info for lastconnected is only updated when the user connects again or something, not necessarily when the user leaves, or vice versa?
It's been some time since I've ran into this, but there's a particular time when it's calculated.

Also, there's several other weird "bugs" or corner cases that Redeemer (numma_cway on the forums) has documented and comments on:
- http://yat.qa/resources (2nd section, "unofficial TeamSpeak 3 docs")

However, unfortunately, this is not something we can fix from the PHP framework.

**Comment by realKitsune** [comment]
Yes, the lastconnected value seems to be more like a "last connection" than "last online" that I expected.

**Comment by ronindesign** [comment]
@realKitsune FYI TS3 Server beta v3.3.0 has fix for this.

From [changelog](https://forum.teamspeak.com/threads/136231-PreRelease-TeamSpeak-3-Server-3-3-0-Beta):
> Fixed: As some have already noted, the last release broke client_lastconnected, so we unbroke it for this release.

**Comment by realKitsune** [comment]
Thanks, will test it tonight, that will conclude my web admin panel :)
